### PR TITLE
Support PHP 8.0

### PR DESCRIPTION
**Pull request by craig410**

### CODE REVIEW

File: src/FestivalsApiClient.php [hunk 1]
```diff
@@ -177,11 +126,9 @@ protected function getSignedUrl(string $url): string
     }
 
     /**
-     * @param BadResponseException $e
-     *
      * @throws FestivalsApiClientException
      */
-    protected function handleApiError($e): void
+    protected function handleApiError(BadResponseException $e): void
```
Comment (acoulton) [inline src/FestivalsApiClient.php:131]: I think this is a BC break, since the method is protected?

File: src/FestivalsApiClientException.php [hunk 2]
```diff
@@ -13,45 +13,29 @@
...
     }
 
-    /**
-     * @param int            $code
-     * @param Exception|null $previous_exception
-     *
-     * @return FestivalsApiClientException
-     */
-    public static function invalidJsonResponse(int $code, $previous_exception = NULL): FestivalsApiClientException
+    public static function invalidJsonResponse(int $code, ?Exception $previous_exception = NULL): FestivalsApiClientException
```
Comment (acoulton) [inline src/FestivalsApiClientException.php:29]: Also a BC break?

**Review by acoulton — COMMENTED** [review]
Looks good, couple of BC breaks on method signatures tho (and off the top of my head I'm not certain what the inheritance/type-compat rules are for adding types to protected properties).

Might be best to ship as a 2.0, as a breaking-but-probably-not-really-breaking release, just in case anyone's extended any of those classes?

**Comment by craig410** [comment]
Yeah I know it's breaking. Should have said or created and targeted the new
branch first. Seems might as well utilise the benefits of supporting the
newer versions.

**Comment by acoulton** [comment]
100% agree, breaking release it is.

**Comment by craig410** [comment]
Equally none of the breaking changes are required to support php 8 so I
could release it as a minor and put the breaks through as a separate pull
and release and then you have the choice as a user

**Comment by acoulton** [comment]
Nah, not worth it when the changeset is so tiny I think.

**Pull request merged**